controlplane/eks/api/v1beta2/awsmanagedcontrolplane_webhook.go

```diff
@@ -426,6 +427,11 @@ func (r *AWSManagedControlPlane) validateNetwork() field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(ipamPoolField, r.Spec.NetworkSpec.VPC.IPv6.IPAMPool, "ipamPool must have either id or name"))
 	}
 
+	if r.Spec.NetworkSpec.VPC.AvailabilityZones != nil && (r.Spec.NetworkSpec.VPC.AvailabilityZoneSelection != nil || r.Spec.NetworkSpec.VPC.AvailabilityZoneUsageLimit != nil) {
```

Comment:
The validation code repeats, so what about using a central location for it.

Similar to these functions:

```
allErrs = append(allErrs, r.Spec.Bastion.Validate()...)
```